syntax = "proto2";

import "db_option.proto";

package proto.db;

message player_hero_data {
	required uint32 id = 1;
}

message player_hero_data_wrap {
	repeated player_hero_data data = 1;
}

message player_hero_patch_data {
	required uint32 id = 1;
}

message player_hero_patch_data_wrap {
	repeated player_hero_patch_data data = 1;
}

message player_hero {
	option (primary_key) = "player_id";
	required uint64 player_id = 1;
	optional player_hero_data_wrap hero_data = 2[(serialize_type) = 1];
	optional player_hero_patch_data_wrap patch_data = 3[(serialize_type) = 1];
}

message player_hero_set {
	repeated player_hero hero_set = 1;
}